### PR TITLE
NFT pinning related UI fixes

### PR DESCRIPTION
**Pull request by cypt4**
Resolves https://github.com/brave/brave-browser/issues/29172
Resolves https://github.com/brave/brave-browser/issues/29486
Resolves https://github.com/brave/brave-browser/issues/29487

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1) Test NFT pinning banner. It should be hidden if the setting is "on" and there is no pinned\in-progress NFTs.
2) Check that NFTs with non-ipfs image url(like https://opensea.io/assets/ethereum/0x70a422c390e8c7682f36611283e8a12254346ee3/1051) are correctly displayed.
3) Test that NFT pinning status is correctly updated while staying on the NFT's detail page.

### CODE REVIEW

File: components/brave_wallet_ui/common/hooks/nft-pin.ts [hunk 1]
```diff
@@ -65,17 +67,27 @@ export function useNftPin () {
 
   const pinningStatusSummary: BraveWallet.TokenPinStatusCode = React.useMemo(() => {
     if (pinnedNftsCount > 0 && inProgressNftCount === 0) {
```
Comment (muliswilliam) [inline components/brave_wallet_ui/common/hooks/nft-pin.ts:69]: `inProgressNftCount` is missing from dependency array.

**Review by muliswilliam — CHANGES_REQUESTED** [review]
All looks good, check the missing dependency comment

**Review by muliswilliam — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.52.36 Chromium: 112.0.5615.49 (Official Build) nightly (64-bit)
-- | --
Revision | `bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}`
OS | Windows 11 Version 22H2 (Build 22621.1555)

- Verified steps from PR description
- Verified NFT pinning banner is hidden if the setting is "on" and there are no pinned\in-progress NFTs.
- Verified NFTs with non-ipfs image like `https://opensea.io/assets/ethereum/0x70a422c390e8c7682f36611283e8a12254346ee3/1051` are correctly displayed but not being pinned
- Verified NFT pinning status is correctly updated while staying on the NFT's detail page.

https://user-images.githubusercontent.com/17010094/231425235-da9ab452-1ee0-48bd-998a-93e91ef1cfa4.mp4

**Pull request merged**